packages/common/src/controllers/submissionController.ts

```diff
@@ -0,0 +1,93 @@
+import { NextFunction, Request, Response } from 'express';
+
+import { Dependencies } from '../config/config.js';
+import submissionService from '../services/submissionService.js';
+import { NotImplemented, getErrorMessage } from '../utils/errors.js';
+import { tsvToJson, validateTsvExtension } from '../utils/fileUtils.js';
+import { BATCH_ERROR_TYPE, BatchError, SubmissionEntity } from '../utils/types.js';
+
+const controller = (dependencies: Dependencies) => {
+	const service = submissionService(dependencies);
+	const { logger } = dependencies;
+	const LOG_MODULE = 'SUBMISSION_CONTROLLER';
+	return {
+		upload: async (req: Request, res: Response, next: NextFunction) => {
+			try {
+				const categoryId = Number(req.params.categoryId);
+				const files = req.files as Express.Multer.File[];
+
+				logger.info(
+					LOG_MODULE,
+					`Upload Submission Request categoryId '${categoryId}' files '${files.map((f) => f.originalname)}'`,
+				);
+
+				const fileErrors: BatchError[] = [];
+				const rawSubmissionEntities: SubmissionEntity[] = [];
+
+				for (const file of files) {
+					try {
```

Comment:
Iterating the files that user sends us, and perform a tsv format validations (valid .tsv extension, parsing tsv to json, ..)